src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -641,10 +642,16 @@ public override void GenerateClassSpecifier(Class @class)
 
             if (@class.NeedsBase)
             {
-                bases.AddRange(
-                    from @base in @class.Bases
-                    where @base.IsClass
-                    select @base.Class.Visit(TypePrinter).Type);
+                foreach (var @base in @class.Bases.Where(b => b.IsClass))
+                {
+                    var typeMaps = new List<System.Type>();
+                    var keys = new List<string>();
+                    this.DisableTypeMap(@base.Class, typeMaps, keys);
+                    var printedBase = @base.Type.Desugar().Visit(TypePrinter);
+                    bases.Add($"{printedBase}{printedBase.NameSuffix}");
+                    for (int i = 0; i < typeMaps.Count; i++)
```

Comment:
Let's add a new property, `TypeMap.Enabled`, or `Active` so we do not have to mutate the lists while generating code. It's wasteful and not very elegant.